.github/workflows/code_checks.yml

```diff
@@ -185,18 +257,11 @@ jobs:
         if: failure()
         uses: actions/upload-artifact@v4
         with:
-          name: Test Artifacts (rails tmp directory)
+          name: Test Artifacts (rails tmp directory) Group ${{ matrix.test_group }}
           path: tmp
           if-no-files-found: ignore
           retention-days: 14
 
-      - name: Remove Test Failure label
-        if: ${{ success() && github.event_name == 'pull_request' }}
-        uses: actions-ecosystem/action-remove-labels@v1
```

Comment:
Test failure has to be done on the group aggregate; Publish Test Results to GitHub is our follow up job that aggregates the results.